frame/ethereum/src/lib.rs

```diff
@@ -559,12 +561,20 @@ impl<T: Config> Pallet<T> {
 
 		// EIP-3607: https://eips.ethereum.org/EIPS/eip-3607
 		// Do not allow transactions for which `tx.sender` has any code deployed.
+		// Exception: Allow transactions from EOAs whose code is a valid delegation indicator (0xef0100 || address).
 		//
 		// This check should be done on the transaction validation (here) **and**
 		// on transaction execution, otherwise a contract tx will be included in
 		// the mempool and pollute the mempool forever.
-		if !pallet_evm::AccountCodes::<T>::get(origin).is_empty() {
-			return Err(InvalidTransaction::BadSigner.into());
+		let origin_code = pallet_evm::AccountCodes::<T>::get(origin);
+		if !origin_code.is_empty() {
+			// Check if code is a valid delegation indicator: 0xef0100 + 20-byte address
+			let is_delegation_indicator =
+				origin_code.len() == 23 && origin_code[0..3] == EIP7702_DELEGATION_INDICATOR;
```

Comment:
Lets move `23` to a constant to some common place. It is being used in other places

```rs
const ADDRESS_SIZE: usize = 20;
const EIP7702_DELEGATION_CODE_SIZE: usize = EIP7702_DELEGATION_INDICATOR.len() + ADDRESS_SIZE;
```

```suggestion
				origin_code.len() == EIP7702_DELEGATION_CODE_SIZE && origin_code[0..3] == EIP7702_DELEGATION_INDICATOR;
```

Reply:
Addressed in https://github.com/polkadot-evm/frontier/pull/1727/commits/fc7adaf0edaedb1b0591a802dfd339725b7e1a99